genesis/sensors/raycaster/raycaster.py

```diff
@@ -0,0 +1,589 @@
+                hit_point = dist * ti_normalize(
+                    ti.math.vec3(
+                        ray_directions[i_p, 0],
+                        ray_directions[i_p, 1],
+                        ray_directions[i_p, 2],
+                    )
+                )
+                output_hits[i_b, i_p * 3 + 0] = hit_point.x
+                output_hits[i_b, i_p * 3 + 1] = hit_point.y
+                output_hits[i_b, i_p * 3 + 2] = hit_point.z
+
+        else:
+            output_hits[i_b, i_p * 3 + 0] = 0.0
+            output_hits[i_b, i_p * 3 + 1] = 0.0
+            output_hits[i_b, i_p * 3 + 2] = 0.0
+            output_hits[i_b, n_points * 3 + i_p] = no_hit_values[i_s]
+
+
+class RaycasterOptions(RigidSensorOptionsMixin, SensorOptions):
+    """
+    Raycaster sensor that performs ray casting to get distance measurements and point clouds.
+
+    Parameters
+    ----------
+    entity_idx : int
+        The global entity index of the RigidEntity to which this sensor is attached.
+    link_idx_local : int, optional
+        The local index of the RigidLink of the RigidEntity to which this sensor is attached.
+    pos_offset : tuple[float, float, float], optional
+        The mounting offset position of the sensor in the world frame. Defaults to (0.0, 0.0, 0.0).
+    euler_offset : tuple[float, float, float], optional
+        The mounting offset quaternion of the sensor in the world frame. Defaults to (0.0, 0.0, 0.0).
+    pattern: RaycastPatternOptions | str
+        The raycasting pattern configuration for the sensor. If a string is provided, a config file with the same name
+        is expected in the `configs` directory.
+    min_range : float, optional
+        The minimum sensing range in meters. Defaults to 0.0.
+    max_range : float, optional
+        The maximum sensing range in meters. Defaults to 20.0.
+    no_hit_value : float, optional
+        The value to return for no hit. Defaults to max_range if not specified.
+    return_world_frame : bool, optional
+        Whether to return points in the world frame. Defaults to False (local frame).
+    only_cast_fixed : bool, optional
+        Whether to only cast rays on fixed geoms. Defaults to False. This is a shared option, so the value of this
+        option for the **first** lidar sensor will be the behavior for **all** Lidar sensors.
+    delay : float, optional
+        The delay in seconds before the sensor data is read.
+    update_ground_truth_only : bool, optional
+        If True, the sensor will only update the ground truth cache, and not the measured cache.
+    draw_debug : bool, optional
+        If True and the rasterizer visualization is active, spheres will be drawn at every hit point.
+    debug_sphere_radius: float, optional
+        The radius of each debug hit point sphere drawn in the scene. Defaults to 0.02.
+    """
+
+    pattern: RaycastPattern | str
+    min_range: float = 0.0
+    max_range: float = 20.0
+    no_hit_value: float | None = None
```

Comment:
As mentioned before, ideally you would like to enforce strict typing, with runtime default eventually.

Reply:
For instance:
```python
@dataclass(init=False)
class Example:
    a: int = 1
    b: int

    def __init__(self, a: int, b: int | None = None):
        self.a = a
        # allow user to override b, else default to a
        self.b = b if b is not None else a
```